### PR TITLE
633: Support for redirect binding signature check using query param v…

### PR DESCRIPTION
**Pull request by vkadam**
This is pull-request to fix issue number #633

### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Have you added an explanation of what problem you are trying to solve with this PR?
* [x] Have you added information on what your changes do and why you chose this as your solution?
* [x] Have you written new tests for your changes?
* [x] Does your submission pass tests?
* [x] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: src/saml2/request.py [hunk 1]
```diff
@@ -37,20 +39,56 @@ def _clear(self):
...
-            pass
-
+               only_valid_cert=False, relayState=None, sigalg=None, signature=None):
         # own copy
         self.xmlstr = xmldata[:]
-        logger.debug("xmlstr: %s", self.xmlstr)
+        logger.debug("xmlstr: %s, relayState: %s, sigalg: %s, signature: %s",
+                     self.xmlstr, relayState, sigalg, signature)
         try:
+            # If redirect binding, and provided SigAlg, Signature use that to verify
```
[the remark this answers is not in the document]
Comment (vkadam) [reply src/saml2/request.py]: Make sense about try/catch, just pushed refactored code, adjusted try/catch and made more readable (I think)

**Comment by vkadam** [comment]
Any updates on this PR would be appreciated. Thanks in advance.

**Comment by c00kiemon5ter** [comment]
hello everyone and thanks for the PR and your patience. I will try to review this ASAP and merge it in order to create the next pysaml2 release.

**Comment by c00kiemon5ter** [comment]
Merged by 718cf98

**Pull request closed without merging**